biofefi/machine_learning/nn_models.py

```diff
@@ -0,0 +1,197 @@
+import numpy as np
+import pandas as pd
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+from sklearn.base import BaseEstimator, ClassifierMixin, RegressorMixin
+
+from biofefi.machine_learning.nn_networks import BaseNetwork
+from biofefi.options.enums import ModelNames, OptimiserTypes, ProblemTypes
+from biofefi.options.ml import BrnnOptions
+from biofefi.utils.custom_loss import compute_brnn_loss
+
+
+class BayesianRegularisedNN(
+    BaseNetwork, BaseEstimator, ClassifierMixin, RegressorMixin
+):
```

Comment:
I don't like this one class doing both regression and classification. I think there should be a regressor class and a classifier class. This will keep the code simpler. 

What I would like to see is: for regressor and classifier, 2 classes. One class will be the underlying neural network. The other will derive either the `ClassifierMixin` or `RegressorMixin`, as well as the `BaseEstimator`. In the latter class you then override the `fit` method to perform the training process of the former class. Message me if you need to explain more.

Reply:
Ok Dan